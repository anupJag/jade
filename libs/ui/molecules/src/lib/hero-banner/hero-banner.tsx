import React, { FC } from 'react';
import { Link } from '@jade/ui/atoms';
import { Box, Image } from '@jade/ui/atoms';

export type HeroBannerProps = {
  imgUrl: string;
  imgAlt: string;
  linkUrl: string;
  height: string;
};

export const HeroBanner: FC<HeroBannerProps> = ({ imgUrl, imgAlt, linkUrl, height }) => {
  return (
    <Box justifyContent="center" d="flex">
      <Link href={linkUrl}>
        <a>
          <Image src={imgUrl} alt={imgAlt} htmlHeight={height} height={height} objectFit="cover" />
        </a>
      </Link>
    </Box>
  );
};
