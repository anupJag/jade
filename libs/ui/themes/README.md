# fe-themes

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `ng test fe-themes` to execute the unit tests via [Jest](https://jestjs.io).
