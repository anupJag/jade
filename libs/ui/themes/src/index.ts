export * from './lib/defaultTheme';
export * from './lib/styled';
export * from './lib/themeProvider';
export { default as styled } from './lib/styled';
export * from './lib/GlobalStyles';
