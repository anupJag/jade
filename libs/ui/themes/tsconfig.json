{
  "extends": "../../../tsconfig.json",
  "compilerOptions": {
    "jsx": "react",
    "allowJs": false,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "types": ["node", "jest"]
  },
  "include": ["**/*.js", "**/*.jsx", "**/*.ts", "**/*.tsx"]
}
