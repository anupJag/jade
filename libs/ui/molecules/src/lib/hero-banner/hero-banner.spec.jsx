import React from 'react';
import { render } from '@testing-library/react';

import { ThemeProvider } from '@jade/ui/themes';
import { HeroBanner } from './hero-banner';

describe('<HeroBanner />', () => {
  let HeroBannerComponent;
  beforeEach(() => {
    HeroBannerComponent = render(
      <ThemeProvider>
        <HeroBanner />
      </ThemeProvider>,
    );
  });

  /**
   * Capturing Sanpshot : Hero Banner
   *
   *  To check style / structure changes in Styled Components
   */
  test('snapshot check', () => {
    const { asFragment } = HeroBannerComponent;
    expect(asFragment()).toMatchSnapshot();
  });
});
