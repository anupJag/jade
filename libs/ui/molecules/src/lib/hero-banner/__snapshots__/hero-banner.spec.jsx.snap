// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeroBanner /> snapshot check 1`] = `
<DocumentFragment>
  <div
    class="css-6lwiu8"
  >
    <a>
      <img
        class="css-1phd9a0"
      />
    </a>
  </div>
</DocumentFragment>
`;
